pkg/backoff/retry.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backoff
+
+import (
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/cenkalti/backoff"
+
+	"github.com/vmware/vic/pkg/trace"
+)
+
+const (
+	// Random numbers generated by fair dice roll :)
+	DefaultInitialInterval     = 10 * time.Millisecond
+	DefaultRandomizationFactor = 0.5
+	DefaultMultiplier          = 10
+	DefaultMaxInterval         = 30 * time.Second
+	DefaultMaxElapsedTime      = 1 * time.Minute
+)
+
+// Retry retries the given function DefaultRetryCount times while sleeping some time between unsuccesfull attempts
+// if retryOnError return true, continue retry, otherwise, return error
+func Retry(operation func() error, retryOnError func(err error) bool) error {
```

Comment:
This is a pretty straightforward change. And is used by other functions. I don't think simply copy/paste is good idea.